libpod/container.go

```diff
@@ -375,6 +375,8 @@ type ContainerConfig struct {
 	// NoCgroups indicates that the container will not create CGroups. It is
 	// incompatible with CgroupParent.
 	NoCgroups bool `json:"noCgroups,omitempty"`
```

Comment:
In retrospect we should have made this something other than a bool, but I think we're too late to fix that. The new question is whether we should make the new value a string instead - CgroupMode, maybe? - to allow for future expansion

Reply:
Can you mark this as deprecated while you're in here? So we know not to use it more